### PR TITLE
feat(sound): add getExternalUrl function

### PR DESCRIPTION
**Pull request by mmackz**
- adds `getExternalUrl` function to sound plugin

This function is used to generate a external url on boost creation.

- [x] https://github.com/rabbitholegg/questdk-plugins/pull/478

Fixes BOOST-4265

https://linear.app/rh-app/issue/BOOST-4265/implement-getexternalurl-in-plugins-sound

### CODE REVIEW

File: packages/soundxyz/src/Soundxyz.test.ts [hunk 1]
```diff
@@ -201,3 +204,14 @@ describe('simulateMint function', () => {
...
 })
+
+describe('getExternalLink', () => {
+  test('should return the correct link', async () => {
+    const contractAddress: Address =
+      '0xE39Df1AD806e84de47D0F6ddf56a0007C678597c'
+    const mintParams = { contractAddress, chainId: Chains.BASE }
+
+    const link = await getExternalUrl(mintParams)
+    expect(link).equals('https://www.sound.xyz/33below/midnight-diner-ii')
```
Comment (jonathandiep) [inline packages/soundxyz/src/Soundxyz.test.ts:215]: Is there a referral query param for Sound?

Comment (mmackz) [reply packages/soundxyz/src/Soundxyz.test.ts:215]: There is. Good call. Will update asap

File: packages/soundxyz/src/Soundxyz.test.ts [hunk 2]
```diff
@@ -175,6 +176,7 @@ describe('simulateMint function', () => {
       contractAddress: '0xdf71F2F15bCcDC7c7A89F01dd45cDE5A43F7e79f',
       amount: BigInt(1),
       recipient: '0xf70da97812CB96acDF810712Aa562db8dfA3dbEF',
+      referralAddress: '0xe3bba2a4f8e0f5c32ef5097f988a4d88075c8b48',
```
Comment (mmackz) [inline packages/soundxyz/src/Soundxyz.test.ts:179]: need to switch this to just `referral`

File: packages/soundxyz/src/Soundxyz.test.ts [hunk 3]
```diff
@@ -201,3 +204,14 @@ describe('simulateMint function', () => {
     expect(request.value).toBe(value)
   })
 })
+
+describe('getExternalLink', () => {
```
Comment (mmackz) [inline packages/soundxyz/src/Soundxyz.test.ts:208]: getExternalUrl

**Comment by mmackz** [comment]
@jonathandiep I have made the recommended changes (added the referral param in external url)

Could you take another pass when you get some time. 🙏

**Review by jonathandiep — APPROVED** [review]

**Pull request merged**